spec/lib/danger/helpers/comments_helper_spec.rb

```diff
@@ -436,6 +438,74 @@ class Dummy
     end
   end
 
+  describe "#generate_message_group_comment" do
+    subject do
+      dummy.generate_message_group_comment(message_group: message_group,
+                                           danger_id: danger_id,
+                                           resolved: resolved,
+                                           template: template)
+    end
+
+    let(:message_group) { Danger::MessageGroup.new(file: file, line: line) }
+    let(:file) { nil }
+    let(:line) { nil }
+    let(:resolved) { [] }
+    let(:danger_id) { Base64.encode64(Random.new.bytes(10)).chomp }
+
+    context "when template is bitbucket_server_message_group" do
+      let(:template) { "bitbucket_server_message_group" }
+
+      context "with one of each type of message" do
+        before do
+          message_group << Danger::Violation.new("Hello!", false, file, line, type: :error)
+          message_group << Danger::Violation.new("World!", false, file, line, type: :warning)
+          message_group << Danger::Violation.new("HOW R", false, file, line, type: :message)
+          message_group << Danger::Markdown.new("U DOING?", file, line)
+        end
+
+        it "spits out a beautiful comment" do
+          expect(subject).to eq <<~COMMENT
+            :no_entry_sign: Hello!
+
+            :warning: World!
+
+            :book: HOW R
+
+
+            U DOING?
+
+            Generated by :no_entry_sign: [Danger](https://danger.systems/ "generated_by_#{danger_id}")
+          COMMENT
+        end
+      end
+
+      context "with two markdowns" do
+        before do
+          message_group << Danger::Markdown.new("markdown one", file, line)
+          message_group << Danger::Markdown.new("markdown two", file, line)
+        end
+
+        it "spits out a beautiful comment" do
+          expect(subject).to eq <<~COMMENT
+            markdown one
+
+            markdown two
+
+            Generated by :no_entry_sign: [Danger](https://danger.systems/ "generated_by_#{danger_id}")
+          COMMENT
+        end
+      end
+
+      context "without any messages" do
+        it "outputs just the Generated line" do
+          expect(subject).to eq <<~COMMENT
+            Generated by :no_entry_sign: [Danger](https://danger.systems/ "generated_by_#{danger_id}")
+          COMMENT
+        end
```

Comment:
Is there value in this?

Reply:
I guess I can't see a way in which this could happen - but better to not crash if that somehow does 👍

Reply:
I may have been getting carried away. I think the idea was "If we write a test with what currently happens, at least we'll know if it ever changes"